tests/virt_autotest/reset_partition.pm

```diff
@@ -0,0 +1,112 @@
+#
+package reset_partition;
+# Summary: reset_partition: Ensure and reset hard disk partition for virt_autotest.
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base "virt_feature_test_base";
+use strict;
+use warnings;
+use testapi;
+use virt_utils;
+use ipmi_backend_utils;
+use Utils::Architectures;
+use utils qw(zypper_call);
+use version_utils 'is_sle';
+use virt_autotest::utils qw(is_xen_host);
+
+sub reset_partition {
+    my ($libvirt_disk) = @_;
+    my $new_disk       = "";
+    my $virt_point     = "/var/lib/libvirt/";
+
+    # Install requested pkgs
+    my @pkgs = qw/xfsprogs coreutils util-linux/;
+    zypper_call "in @pkgs";
+
+    # Get some debug info about hard disk topology
+    record_info("Report file system disk space", script_output("df"));
+    record_info("Find mounted filesystem",       script_output("findmnt"));
+
+    # Check with all existed Hard disks
+    my @disks = split(/\n/, script_output("lsblk -n -l -o NAME -d -e 7,11"));
+    my $dev   = "/dev/";
+    foreach my $disk (@disks) {
+        assert_script_run('file ' . $dev . $disk);
+    }
+
+    # Assign a New FULL Disk for /var/lib/libvirt
+    foreach (0 .. $#disks) {
+        if ($libvirt_disk eq $disks[$_]) {
+            $new_disk = $new_disk . $dev . $disks[int($_ + 1)];
+            last;
+        } else {
+            $new_disk = $new_disk . $dev . $disks[0];
+            last;
+        }
+    }
+    record_info('Assign a New Disk for /var/lib/libvirt:', $new_disk);
+
+    # Wipe Hard Disk Clean via dd for assigned a new full disk
+    assert_script_run("dd if=/dev/zero of=$new_disk count=1M", timeout => 1500, fail_message => "Failed to wipe hard disk clean on $new_disk");
+
+    # Create New Disk Partition
+    assert_script_run 'echo -e "g\nn\n\n\n\nt\n8e\np\nw" | fdisk ' . $new_disk;
+
+    # Format New Disk Partition
+    $new_disk .= 1;
+    assert_script_run "mkfs.xfs -f $new_disk";
+
+    # Umount existed /var/lib/libvirt mounted point
+    assert_script_run "umount -f -l $virt_point";
+
+    # Wipe existed /var/lib/libvirt mounted point from /etc/fstab
+    script_run "sed -i '/\\/var\\/lib\\/libvirt/d' /etc/fstab";
+
+    # Add a new /var/lib/libvirt mounted point to /etc/fstab
+    assert_script_run "echo '$new_disk $virt_point xfs defaults 0 0' >> /etc/fstab";
+
+    # Mount the new /var/lib/libvirt mounted point
+    assert_script_run "mount -a";
+
+    my @dirs = qw/boot dnsmasq filesystems images network qemu swtpm/;
+    push @dirs, qw(libxl lxc) if (is_xen_host);
+    foreach my $dir (@dirs) {
+        assert_script_run('mkdir -p ' . $virt_point . $dir);
+    }
+
+    # Check with new /var/lib/libvirt mounted point
+    record_info("System MOUNT Info",       script_output("mount"));
+    record_info("List Block Devices",      script_output("lsblk"));
+    record_info("Ensure /var/lib/libvirt", script_output("ls -al /var/lib/libvirt"));
+
+}
+
+sub run {
+    my $self = shift;
+
+    return if (is_sle("<15") or !is_x86_64);
```

Comment:
Why is this only applied for sle15+? Should not it be applied for all products(sle12/sle15)?

Reply:
Good question, I had do more investigation here. 
1st, I had reviewed all current osd virt_auto results, figure out the problem as no enough pool space left for the volume allocated during virt-clone just only occurred on sles15sp3 kvm| xen without sles12+
2nd, regarding the partition to store guest image both on sles12+ and sles15+:
sles15+: e.g sles15sp2, sles15sp3
created the partition as /var/lib/libvirt partition, format as xfs
sles12+: e.g sles12sp4, sles12sp5
created the partition as /var/lib/libvrit/images, format as brtfs
thus, there was used the different partition and format disk style to store guest image both on sles12+ and sles15+. 
So, just realize reset new disk as new libvirt mounted point for virtual network test to deal with the problem no enough pool space left for the volume allocated during virt-clone on sles15+ without sles12+ for this PR. 
Moreover, I will create new PR to fix them if found the same problem was occurred on sles12+.

Reply:
I see. It is totally fine to enable it  for sle15 only, because sle12 does not hit the problem and once it does, add support for sle12.
Would you please add a comment before L95 to explain the reason?

Reply:
sure, had added comments to explain the reason now.